src/evo/netinfo.h

```diff
@@ -246,7 +246,7 @@ class MnNetInfo final : public NetInfoInterface
     }
 
     NetInfoStatus AddEntry(const uint8_t purpose, const std::string& service) override;
-    NetInfoList GetEntries() const override;
+    NetInfoList GetEntries(std::optional<uint8_t> purpose_opt = std::nullopt) const override;
```

Comment:
std::optional here looks a bit too complicated; maybe better to add one more _fake_ value to enum such as `Purpose::ALL` ?